webui/react/src/pages/TaskList.tsx

```diff
@@ -88,6 +93,36 @@ const TaskList: React.FC = () => {
     return false;
   }, [ selectedTasks ]);
 
+  const fetchTasks = useCallback((): void => {
+    requestCommands({});
+    requestNotebooks({});
+    requestShells({});
+    requestTensorboards({});
+  }, [ requestCommands, requestNotebooks, requestShells, requestTensorboards ]);
+
+  /*
+   * Check once every second to see if all task endpoints have resolved.
+   * Consider it a failure if the number of checks exceed 10 times.
+   */
+  const checkForTaskListUpdate = useCallback((): Promise<void> => {
```

Comment:
suggestion: this is a common logic we could have separately. something like `waitUntil(predicate | async predicate, timeout, check frequency)`

Reply:
probably two separate versions of sync and async

Reply:
Great idea. I don't know if I like this approach of waiting and 
Created a [separate ticket](https://determinedai.atlassian.net/browse/DET-3715) to keep track of this work.